torch/csrc/jit/register_string_ops.cpp

```diff
@@ -547,48 +547,94 @@ auto reg_str_ops_2 = torch::RegisterOperators()
            post_partition);
         })
 
-    .op("aten::split(str self, str separator=' ', int max=-1) -> str[]",
+    // Due to type constraints in TorchScript, we use a sentinel string
+    // "__DO_NOT_USE" as the default separator for split functions.
+    // This denotes splitting by any whitespace, or default behavior when
+    // the separator is not specified
+    .op("aten::split(str self, str separator='__DO_NOT_USE', int max=-1) -> str[]",
         [](std::string string, std::string separator, int64_t max) {
+          TORCH_CHECK(separator != "", "empty separator");
+          std::string default_separator = "__DO_NOT_USE"; 
           std::string::size_type prev_pos = 0;
           std::string::size_type pos = 0;
           c10::List<std::string> splits;
           auto count = 0;
-          while ((pos = string.find(separator, pos)) != std::string::npos) {
-            count++;
-            if (max >= 0 && count > max) {
-              break;
+          std::vector<std::string> separators = {separator};
+          if (separator == default_separator) {
+            separators = { "\n", "\t", " " };
+          }
+          do {
+            auto min_pos = std::string::npos;
```

Comment:
there's a lot of code copy paste between here and rsplit, could they be combined?